File: domain/src/main/java/com/kaciula/archiman/domain/util/TrampolineSchedulerProvider.java
package com.kaciula.archiman.domain.util;

import io.reactivex.Scheduler;
import io.reactivex.annotations.NonNull;
import io.reactivex.schedulers.Schedulers;

/**
 * Implementation of the {@link SchedulerProvider} making all {@link Scheduler}s trampoline.
 */
public class TrampolineSchedulerProvider implements SchedulerProvider {

  @NonNull
  @Override
  public Scheduler computation() {
    return Schedulers.trampoline();
  }

  @NonNull
  @Override
  public Scheduler io() {
    return Schedulers.trampoline();
  }

  @NonNull
  @Override
  public Scheduler ui() {
    return Schedulers.trampoline();
  }
}

File: domain/src/main/java/com/kaciula/archiman/domain/util/TrampolineSchedulerProvider.kt
package com.kaciula.archiman.domain.util

import io.reactivex.Scheduler
import io.reactivex.schedulers.Schedulers

/**
 * Implementation of the [SchedulerProvider] making all [Scheduler]s trampoline.
 */
class TrampolineSchedulerProvider : SchedulerProvider {

    override fun computation(): Scheduler = Schedulers.trampoline()

    override fun io(): Scheduler = Schedulers.trampoline()

    override fun ui(): Scheduler = Schedulers.trampoline()
}
